lib/object-adapter.js

```diff
@@ -1,12 +1,35 @@
-var ObjectAdapter = function() {};
-ObjectAdapter.prototype.build = function(Model, props) {
-  return props;
-};
-ObjectAdapter.prototype.save = function(doc, Model, cb) {
-  process.nextTick(function() {
-    cb(null, doc);
-  });
-};
-ObjectAdapter.prototype.destroy = function(doc, Model, cb) {};
-
-module.exports = ObjectAdapter;
+(function(adapter) {
+
+  if (typeof exports !== 'undefined') {
+    var ObjectAdapter = adapter();
+    if (typeof module !== 'undefined' && module.exports) {
+      exports = module.exports = ObjectAdapter;
+    }
+    exports.ObjectAdapter = ObjectAdapter;
```

Comment:
Yeah okay, we shouldn't need that. That will create ObjectAdapter.ObjectAdapter, which we don't want. I would just have:

```
if (typeof module !== 'undefined) {
  var ObjectAdapter = adapter();
  module.exports = ObjectAdapter;
}
```

That will cover the Node.js use case, and then the define.amd will cover the AMD use case.